.github/actions/fullnode-sync/action.yaml

```diff
@@ -0,0 +1,85 @@
+# This action runs a public fullnode using a specified branch ($BRANCH),
```

Comment:
nit: I would make the environment variables `BRANCH`, `NETWORK`, `BOOTSTRAPPING_MODE`, etc into action inputs, which makes it slightly more modular and clearer to the user what's required. Also, it makes it easier later on to move all these actions to a separate repo! Here's an example (from me moving out the replay-verify job) : https://github.com/aptos-labs/actions/blob/0a510a8f270c9b5093c3e79043f4613dbb07aeb3/replay-verify/action.yml